helix-term/src/ui/picker.rs

```diff
@@ -61,10 +61,36 @@ impl<T> FilePicker<T> {
     }
 
     fn calculate_preview(&mut self, editor: &Editor) {
+        /// Biggest file size to preview in bytes
+        const MAX_BYTE_PREVIEW: u64 = 10 * 1024 * 1024;
+
         if let Some((path, _line)) = self.current_file(editor) {
             if !self.preview_cache.contains_key(&path) && editor.document_by_path(&path).is_none() {
+                let metadata = std::fs::File::open(&path).unwrap().metadata().unwrap();
+                let content_type = {
+                    let mut buffer = Vec::new();
+                    let file = std::fs::File::open(&path).unwrap();
+                    file.take(MAX_BYTE_PREVIEW)
+                        .read_to_end(&mut buffer)
+                        .unwrap();
+                    content_inspector::inspect(&buffer)
+                };
```

Comment:
Since content_inspector can check with only 1024 bytes, I keep read twice on 59de05f.

Should I buffer first 1024 bytes? ( I think this will make code complicated a bit)

Reply:
I'll refactor it, the `unwrap`s all need to be removed.  1024 bytes is fine

Reply:
Thank you, I forgot to remove unwrap